src/components/calendar.rs

```diff
@@ -1,28 +1,125 @@
+                            div {
+                                (sec.subject_code) " " (sec.course_code)
+                            }
+                            div {
+                                (sec.sequence_code)
+                            }
+                            div {
+                                (sec.enrollment) "/" (sec.enrollment_capacity) " (enrolment)"
+                            }
+                            div {
+                                (sec.waitlist) "/" (sec.waitlist_capacity) " (waitlist)"
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    )
+}
 
+fn render_day(day: Day, timeslots: &Vec<Time>, sections: &Vec<&Section>) -> Markup {
+    let earliest = timeslots.first().unwrap();
+    let latest = timeslots.last().unwrap();
+    debug!(?day, ?sections);
     html!(
-        div id="calendar" class="w-full h-full flex" {
-            div class="w-12 flex flex-col" {
-                div class="text-[0.5rem] border-b-2 box-content grow" { "time" }
-                @for time in &timeslots {
-                    div class="text-[0.25rem] border-b-2 box-content grow" { (time) }
+        div class="flex-1 flex flex-col" {
+            div class="text-[calc(1.5vh)] lg:text-sm shrink flex justify-center items-center" { (day.to_string().to_lowercase()) }
+            div class="relative flex flex-col grow gap-0.5 lg:gap-1" {
+                @for _ in timeslots {
+                    div class="h-auto grow bg-neutral-100 dark:bg-neutral-600" {  }
+                }
+                @for section in sections {
+                    (render_section_cards(earliest, latest, &section, day))
                 }
             }
-            @for day in &days {
-                div class="grow flex flex-col" {
-                    div class="text-[0.5rem] border-b-2 box-content grow flex justify-center items-center" { (day) }
-                    @for _ in &timeslots {
-                        div class="text-[0.25rem] border-b-2 box-content grow" { "." }
+        }
+    )
+}
+
+pub fn render(sections: &Vec<&Section>) -> Markup {
+    debug!(?sections);
+
+    let meeting_times: Vec<&MeetingTime> = sections.iter().flat_map(|s| &s.meeting_times).collect();
+    debug!(?meeting_times);
+
+    let earliest: Time = meeting_times
+        .iter()
+        .flat_map(|mt| mt.start_time)
+        .min()
+        .unwrap_or(jiff::civil::time(8, 30, 0, 0));
+
+    let latest = meeting_times
+        .iter()
+        .flat_map(|mt| mt.end_time)
+        .max()
+        .unwrap_or(jiff::civil::time(23, 20, 0, 0))
+        - 20.minutes();
+
+    debug!(?earliest, ?latest);
+
+    let timeslots = earliest
+        .series(30.minutes())
+        .take_while(|&t| t <= latest)
+        .collect::<Vec<Time>>();
+
+    let saturday = meeting_times
+        .iter()
+        .filter(|mt| Day::Saturday.is_in_days(mt.days))
+        .count()
+        > 0;
```

Comment:
i can't query the db but if there's never a meeting on sunday then i don't see why we'd ever render it. i am willing to add code to render the sunday column **if** there is a meeting on that day but only for the purpose of backwards-compat. no need to waste screen space otherwise